elastic/kibana
# [Uptime] Fix ML job/rule edit error

## Summary

This PR attempts to fix an error encountered when editing an existing ML rule in Uptime UI.


## Testing

You can reproduce this issue in latest `main` by creating a monitor (use synthetics or uptime, doesn't matter) and open the legacy Uptime UI.

- Navigate to the monitor detail view, and create an ML alert (you might need to engage trial license or other means to activate ML before you can do this.)
- Create your alert
- Edit the alert (the screen should crash)
- Checkout this branch
- Try to edit again. You should be able to edit and update the rule object multiple times without a crash
- See GIF below for further illustration on repro

**NOTE:** This GIF doesn't correspond exactly to what you can expect to see on `main`, as I recorded it while I was investigating the problem. You should expect the edit flyout not to load at all on `main`, rather than seeing an error toast after attempting to POST like this GIF shows. But the actual flow is exactly the same up to the point of editing the detail flyout.

![20230418185724](https://user-images.githubusercontent.com/18429259/233210656-0a10fa93-f612-4764-afc0-028bc4289741.gif)


### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
